### PR TITLE
Embedded Backfila can create runs taking parameters as POKO

### PR DESCRIPTION
**Pull request by zcross**
The recent change to make parameters representable to backfill implementors as a POKO instead of a map is great! This PR makes a tweak to support specifying parameters as a POKO when creating backfill instances in tests.

### CODE REVIEW

File: client-misk/src/main/kotlin/app/cash/backfila/client/misk/internal/BackfilaParameters.kt [hunk 1]
```diff
@@ -15,9 +14,8 @@ import kotlin.reflect.jvm.jvmErasure
...
 import okio.ByteString.Companion.encodeUtf8
 
-inline fun <reified T : Any> parametersToBytes(parameters: T): Map<String, ByteString> {
-  val parametersClass = T::class
-
+fun parametersToBytes(parameters: Any): Map<String, ByteString> {
+  val parametersClass = parameters::class
   val map = mutableMapOf<String, ByteString>()
 
   for (property in parametersClass.memberProperties) {
```
Comment (zcross) [inline client-misk/src/main/kotlin/app/cash/backfila/client/misk/internal/BackfilaParameters.kt:21]: Without this part of the diff, the added test fails.
Debugging it, it seems due to `parametersClass` always being `Any`, which means this for loop is over an empty `parametersClass.memberProperties`.

File: client-misk/src/test/kotlin/app/cash/backfila/client/misk/hibernate/SinglePartitionHibernateBackfillTest.kt [hunk 2]
```diff
@@ -266,6 +266,18 @@ abstract class SinglePartitionHibernateBackfillTest {
     )
   }
 
+  @Test fun parametersAsPoko() {
```
Comment (shellderp) [inline client-misk/src/test/kotlin/app/cash/backfila/client/misk/hibernate/SinglePartitionHibernateBackfillTest.kt:269]: do people actually say poko? lol

File: client-misk/src/main/kotlin/app/cash/backfila/client/misk/internal/BackfilaParameters.kt (hunk 1 above)
Comment (shellderp) [reply client-misk/src/main/kotlin/app/cash/backfila/client/misk/internal/BackfilaParameters.kt:21]: huh, weird

**Comment by zcross** [comment]
Build failure seems like a flake and I can't reproduce locally:
```
Failures (1):
  JUnit Jupiter:BackfillRunnerTest:numThreadsIncreased()
    MethodSource [className = 'app.cash.backfila.service.BackfillRunnerTest', methodName = 'numThreadsIncreased', methodParameterTypes = '']
    => java.lang.AssertionError: 
Expecting actual not to be null
```
I don't see an option to retry manually in CircleCI, so I guess I'll kick it by rebasing?

**Comment by shellderp** [comment]
> Build failure seems like a flake and I can't reproduce locally:
> 
> ```
> Failures (1):
>   JUnit Jupiter:BackfillRunnerTest:numThreadsIncreased()
>     MethodSource [className = 'app.cash.backfila.service.BackfillRunnerTest', methodName = 'numThreadsIncreased', methodParameterTypes = '']
>     => java.lang.AssertionError: 
> Expecting actual not to be null
> ```
> 
> I don't see an option to retry manually in CircleCI, so I guess I'll kick it by rebasing?
> 
> Next build, another flake:
> 
> ```
> Failures (1):
>   JUnit Jupiter:BackfillRunnerTest:retriesExhaustedPausesBackfill()
>     MethodSource [className = 'app.cash.backfila.service.BackfillRunnerTest', methodName = 'retriesExhaustedPausesBackfill', methodParameterTypes = '']
>     => kotlinx.coroutines.TimeoutCancellationException: Timed out waiting for 2000 ms
> ```

never seen that but maybe circleci is a bit slow. unfortunately coroutines make it hard to test without lame timeouts

**Pull request closed without merging**